### PR TITLE
release: v1.43.0

### PR DESCRIPTION
**Pull request by abernix**
> **Note**
> **This particular PR must be true-merged to `main`.**

* This PR is only ready to review when it is marked as "Ready for Review".  It represents the merge to the `main` branch of an upcoming release (version number in the title).
* It will act as a staging branch until we are ready to finalize the release.
* We may cut any number of alpha and release candidate (RC) versions off this branch prior to formalizing it.
* This PR is **primarily a merge commit**, so reviewing every individual commit shown below is **not necessary** since those have been reviewed in their own PR.  However, things important to review on this PR **once it's marked "Ready for Review"**:
    - Does this PR target the right branch? (usually, `main`)
    - Are the appropriate **version bumps** and **release note edits** in the end of the commit list (or within the last few commits).  In other words, "Did the 'release prep' PR actually land on this branch?"
    - If those things look good, this PR is good to merge!

### CODE REVIEW

File: .changesets/feat_subscription_websocket_heartbeat.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+An example of configuration:
+
+```yaml
+subscription:
+  mode:
+    passthrough:
+      all:
+        path: /graphql
+        heartbeat_interval: enable #Optional
```
Comment (shorgi) [inline .changesets/feat_subscription_websocket_heartbeat.md:13]:
```suggestion
### Support configurable heartbeat for subscriptions using the WebSocket protocol  ([Issue #4621](https://github.com/apollographql/router/issues/4621))

To support GraphQL Subscription WebSocket implementations such as [DGS](https://netflix.github.io/dgs/) that drop idle connections by design, the router adds the ability to configure a heartbeat to keep active connections alive.

An example router configuration:

```yaml 
subscription:
  mode:
    passthrough:
      all:
        path: /graphql
        heartbeat_interval: enable #Optional
```

File: .changesets/fix_bnjjj_multipart_test.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+### Fix chunk formatting in multipart protocol ([Issue #4634](https://github.com/apollographql/router/issues/4634))
+
+This PR changes the way we're sending chunks in the stream. Instead of finishing the chunk with `\r\n` we don't send this at the end of our current chunk but instead at the beginning of the next one. For the end users nothing changes but it let us to close the stream with the right final boundary by appending `--\r\n` directly to the last chunk.
```
Comment (shorgi) [inline .changesets/fix_bnjjj_multipart_test.md:3]:
```suggestion
Previously, when sending a stream of chunks for HTTP multipart, the router finished each chunk by appending it with `\r\n`. 

Now, the router doesn't append `\r\n` to the current chunk but instead prepends it to the next chunk. This enables the router to close a stream with the correct final boundary by appending `--\r\n` directly to the last chunk. 
```

File: .changesets/fix_bryn_zipkin_service_name.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+### Zipkin service name not populated ([Issue #4807](https://github.com/apollographql/router/issues/4807))
+
+Zipkin trace exporter now respects service name configuration from yaml or environment variables.
```
Comment (shorgi) [inline .changesets/fix_bryn_zipkin_service_name.md:3]:
```suggestion
The Zipkin trace exporter now respects service name configuration from YAML or environment variables.
```

File: .changesets/fix_format_response_index_map_with_capacity.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+### use Object::with_capacity with selection set len in format_response ([PR #4775](https://github.com/apollographql/router/pull/4775))
+
+This preallocates output object size in response formatting, bringing a small performance improvement
```
Comment (shorgi) [inline .changesets/fix_format_response_index_map_with_capacity.md:3]:
```suggestion
### Preallocate response formatting output ([PR #4775](https://github.com/apollographql/router/pull/4775))

To improve runtime performance, the router now preallocates its response formatting's output object. 
```

**Review by abernix — COMMENTED** [review]
I've moved all of @shorgi 's edits to https://github.com/apollographql/router/pull/4831 and applied them as cleanly as I could, so marking them all as resolved here.

**Review by garypen — APPROVED** [review]

**Review by Geal — APPROVED** [review]

**Pull request merged**